src/com/spikes2212/genericsubsystems/autonomus/commands/DriveTankWithPID.java

```diff
@@ -0,0 +1,100 @@
+package com.spikes2212.genericsubsystems.autonomus.commands;
+
+import com.spikes2212.genericsubsystems.drivetrains.TankDrivetrain;
+
+import edu.wpi.first.wpilibj.PIDController;
+import edu.wpi.first.wpilibj.command.Command;
+
+public class DriveTankWithPID extends Command {
+	// TODO Auto-generated constructor stub
+
+	public DriveTankWithPID() {
+		// TODO Auto-generated constructor stub
+	}
+
+	private TankDrivetrain tankDrivetrain;
+	private static double KP = 1;
+	private static double KI = 1;
+	private static double KD = 1;
+	private static double tolerance = 1;
+	private PIDController leftMovmentControl;
+	private PIDController rightMovmentControl;
+
+	public static void setP(double P) {
+		KP=P;
+	}
+
+	public static double getP() {
+		return KP;
+	}
+
+	public static void setI(double I) {
+		KI = I;
+	}
+
+	public static double getI() {
+		return KI;
+	}
+
+	public static void setD(double D) {
+		KD = D;
+	}
+
+	public static double getD() {
+		return KD;
+	}
+
+	public static double getTolarance() {
+		return tolerance;
+	}
+
+	public static void setTolarance(double tolarance) {
+		DriveTankWithPID.tolerance = tolarance;
+	}
+
+	public DriveTankWithPID(double setPoint) {
```

Comment:
You don't get a tankDrivetrain anywhere, and you also don't run requires() on it